pkg/kubelet/kubelet.go

```diff
@@ -439,12 +439,20 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	}
 
 	serviceIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	var serviceLister corelisters.ServiceLister
 	if hasValidTPClients(kubeDeps.KubeTPClients) {
-		serviceLW := cache.NewListWatchFromClient(kubeDeps.KubeTPClients[0].CoreV1(), "services", metav1.NamespaceAll, fields.Everything())
-		r := cache.NewReflector(serviceLW, &v1.Service{}, serviceIndexer, 0)
-		go r.Run(wait.NeverStop)
+		serviceLister := make([]corelisters.ServiceLister, len(kubeDeps.KubeTPClients))
+
+		for i := range serviceLister {
```

Comment:
nit: better to use the source length which is the kubeClient's length

Reply:
Yes, in line 444, use len(kubeDeps.KubeTPClients)